crypto_sign/ml-dsa-44/clean/sign.c

```diff
@@ -91,145 +97,160 @@ int PQCLEAN_DILITHIUM2_CLEAN_crypto_sign_signature(uint8_t *sig,
     poly cp;
     shake256incctx state;
 
+    if (ctxlen > 255) {
+        return -1;
+    }
+
     rho = seedbuf;
     tr = rho + SEEDBYTES;
     key = tr + TRBYTES;
     rnd = key + SEEDBYTES;
     mu = rnd + RNDBYTES;
     rhoprime = mu + CRHBYTES;
-    PQCLEAN_DILITHIUM2_CLEAN_unpack_sk(rho, tr, key, &t0, &s1, &s2, sk);
+    PQCLEAN_MLDSA44_CLEAN_unpack_sk(rho, tr, key, &t0, &s1, &s2, sk);
 
-    /* Compute mu = CRH(tr, msg) */
+    /* Compute mu = CRH(tr, 0, ctxlen, ctx, msg) */
+    mu[0] = 0;
+    mu[1] = ctxlen;
```

Comment:
MSVC requires a sacrifice
```suggestion
    mu[1] = (uint8_t)ctxlen;
```